docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,837 @@
+- `applications/<my_application_id>/guilds/<guild_id>/commands/<command_id>` for guild commands
+
+Because Slash Commands have unique names within a scope, we treat `POST` requests for new commands as upserts. That means **making a new command with an already-used name for your application will update the existing command**.
+
+Full documentation of endpoints can be found in [Endpoints](#DOCS_INTERACTIONS_SLASH_COMMANDS/endpoints).
+
+## Receiving an Interaction
+
+When a user completes a Slash Commands, your app will receive an **Interaction**. Your app can receive an interaction in one of two ways:
+
+- Via gateway event, `INTERACTION_CREATE` <docs>
+- Via outgoing webhook
+
+These two methods are **mutually exclusive**; you can _only_ receive Interactions one of the two ways. The `INTERACTION_CREATE` gateway event will be handled by the library you are using, so we'll just cover outgoing webhooks.
+
+In your application in the Developer Portal, there is a field on the main page called "Interactions Endpoint URL". If you want to receive Interactions via outgoing webhook, you can set your URL in this field. In order for the URL to be valid, you must be prepared for two things ahead of time:
+
+> info
+> These steps are only necessary for webhook-based Interactions. It is not required for receiving them over the gateway.
+
+1. Your endpoint must be prepared to ACK a `PING` message
+2. Your endpoint must be set up to properly handle signature headers--more on that in [Security and Authorization](#DOCS_INTERACTIONS_SLASH_COMMANDS/security-and-authorization)
+
+If either of these are not complete, we will not validate your URL and it will fail to save.
+
+When you attempt to save a URL, we will send a `POST` request to that URL with a `PING` payload. The `PING` payload has a `type: 1`. So, to properly ACK the payload, return a `200` reponse with a payload of `type: 1`:
+
+```py
+@app.route('/', methods=['POST'])
+def my_command():
+    if request.json["type"] == 1:        
+        return jsonify({
+            "type": 1
+        })
+```
+
+You'll also need to properly set up [Security and Authorization](#DOCS_INTERACTIONS_SLASH_COMMANDS/security-and-authorization) on your endpoint for the URL to be accepted. Once both of those are complete and your URL has been saved, you can start receiving Interactions via webhook! At this point, your app will **no longer receive Interactions over the gateway**. If you want to receive them over the gateway again, simply delete your URL.
+
+An Interaction includes the `data` that the user sent in the command, as well as some metadata. It looks like this:
+
+```js
+{
+    "type": 2,
+    "token": "A_UNIQUE_TOKEN",
+    "member": { 
+        "user": {
+            "id": 53908232506183680,
+            "username": "Mason",
+            "avatar": "a_d5efa99b3eeaa7dd43acca82f5692432",
+            "discriminator": "1337",
+            "public_flags": 131141
+        },
+        "roles": [539082325061836999],
```

Comment:
Shouldn't this role snowflake be a string? When I looked at the request body I got when executing an interaction, the snowflake was a string.